### PR TITLE
Add offline install instructions for plugins

### PR DESCRIPTION
**Pull request by dadoonet**
Follow up of https://github.com/elastic/elasticsearch/issues/15925#issuecomment-171250150

This commit adds offline install instructions for plugins.

### CODE REVIEW

**Comment by dadoonet** [comment]
@clintongormley Could you review it please?
I plan to apply this PR in 5.0, 5.x. And may be 2.4. WDYT?

File: docs/plugins/analysis-icu.asciidoc [hunk 1]
```diff
@@ -21,6 +21,9 @@ sudo bin/elasticsearch-plugin install analysis-icu
 The plugin must be installed on every node in the cluster, and each node must
 be restarted after installation.
 
+This plugin can be downloaded for offline install from
```
Comment (clintongormley) [inline docs/plugins/analysis-icu.asciidoc]: I would make `offline install` a link to https://www.elastic.co/guide/en/elasticsearch/plugins/current/installation.html#_custom_url_or_file_system - add an `[[id]]` to that section for easy linking.

File: docs/plugins/analysis-icu.asciidoc [hunk 2]
```diff
@@ -21,6 +21,9 @@ sudo bin/elasticsearch-plugin install analysis-icu
...
+{plugin_url}/analysis-icu/{version}/analysis-icu-{version}.zip[elastic download service].
```
Comment (clintongormley) [inline docs/plugins/analysis-icu.asciidoc]: i'd remove `[elastic download service]` and just show the real URL.

File: docs/plugins/index.asciidoc [hunk 3]
```diff
@@ -4,6 +4,7 @@
 :guide:          https://www.elastic.co/guide
 :version:        6.0.0-alpha1
 :lucene_version: 6.2.0
+:plugin_url:     https://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin
```
Comment (clintongormley) [inline docs/plugins/index.asciidoc]: @rjernst will this URL be correct with the unified release?

**Comment by clintongormley** [comment]
thanks @dadoonet - looks good - i've added a couple of comments

**Comment by dadoonet** [comment]
@clintongormley I applied your comments. I build the docs and it looks good.
Please let me know [where we want to merge this PR in](https://github.com/elastic/elasticsearch/pull/20415#issuecomment-246349506).

**Comment by clintongormley** [comment]
@dadoonet 2.4, 5.0, 5.x, master sound good to me

**Pull request merged**